### PR TITLE
Changed usage of `awsprometheusremotewriteexporter` to `prometheusremotewriteexporter` and `sigv4authextension`

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:**
This PR is aligned with the [PR](https://github.com/aws-observability/aws-otel-collector/pull/1066) to add the sigv4 extension to the ADOT Collector. 

**Testing:** 
Note that testing has NOT been able to be performed; see #605 for more information

### CODE REVIEW

**Comment by Aneurysm9** [comment]
Does this leave the `awsprometheusremotewrite` exporter untested?  Can we ensure that both paths are tested?

**Comment by erichsueh3** [comment]
> Does this leave the `awsprometheusremotewrite` exporter untested? Can we ensure that both paths are tested?

Added back in the `awsprometheusremotewrite` exporter tests

**Comment by erichsueh3** [comment]
I have cloned the tests that include the AWS PRW Exporter. Tests that include the AWS PRW Exporter have been renamed to `<original_directory_name>_awsprw`. I have also added in tests for the PRW Exporter + Sigv4 Extension, and those now reside in the original directories that the AWS PRW Exporter tests once lived in. This was all done so the two Exporters were being tested separately.

We have tested the `otlp_grpc_exporter_cw_amp_xray_ecs` test, the `prometheus_static` test, and the `prometheus_mock` test in ECS (`prometheus_mock` in local as well). The `otlp_metric_amp` test is supposed to be tested in EC2, but even with a default Collector image + unchanged tests, it is unable to be tested. The `prometheus_sd`, `prometheus_sd_adot_operator`, and `prometheus_static_adot_operator` tests are supposed to be tested in EKS, but even with a default Collector image + unchanged tests, it is unable to be tested. However, even though there are some environments that we still cannot test in (the above tests can only be tested in either EC2 or EKS), there is high confidence that this suite of tests all pass.

**Comment by erichsueh3** [comment]
The `otlp_metric_amp` test fails in EC2, with this command:
```
cd terraform/ec2 && terraform init && terraform apply -auto-approve \
    -var="aoc_image_repo=public.ecr.aws/aws-observability/aws-otel-collector" \
    -var="aoc_version=latest" \
    -var="testcase=../testcases/otlp_metric_amp_awsprw" \
    -var-file="../testcases/otlp_metric_amp_awsprw/parameters.tfvars"
```
This is the error:
```
Error: remote-exec provisioner error
│ 
│   with null_resource.start_collector[0],
│   on main.tf line 277, in resource "null_resource" "start_collector":
│  277:   provisioner "remote-exec" {
│ 
│ error executing "/tmp/terraform_485573440.sh": Process exited with status 1
```
which stems from this
```
null_resource.start_collector[0] (remote-exec): error: aws-otel-collector.rpm: not an rpm package (or package manifest):
```

The tests in EKS fail with a different error than above, but they all fail the same. With this command:
```
cd terraform/eks && terraform init && terraform apply -auto-approve \
    -var="aoc_image_repo=public.ecr.aws/aws-observability/aws-otel-collector" \
    -var="aoc_version=latest" \
    -var="testcase=../testcases/prometheus_sd" \
    -var-file="../testcases/prometheus_sd/parameters.tfvars" \
    -var="eks_cluster_name=<MY_CLUSTER_NAME>r"
```
This is the error:
```
│ Error: Unauthorized
│ 
│   with kubernetes_namespace.aoc_ns,
│   on main.tf line 99, in resource "kubernetes_namespace" "aoc_ns":
│   99: resource "kubernetes_namespace" "aoc_ns" {
│ 


│ Error: Unauthorized
│ 
│   with kubernetes_namespace.aoc_fargate_ns,
│   on main.tf line 106, in resource "kubernetes_namespace" "aoc_fargate_ns":
│  106: resource "kubernetes_namespace" "aoc_fargate_ns" {
│ 
```

File: terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs/otconfig.tpl [hunk 1]
```diff
@@ -1,6 +1,10 @@
 extensions:
   pprof:
     endpoint: 0.0.0.0:1777
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs/otconfig.tpl:5]: does this need to be? I see it in this format in the exporters section.
```suggestion
    region: '${region}'
```

File: terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs_awsprw/otconfig.tpl [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    namespace: '${otel_service_namespace}/${otel_service_name}'
+    region: '${region}'
+    resource_to_telemetry_conversion:
+      enabled: true
+  awsprometheusremotewrite:
+    endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+    resource_to_telemetry_conversion:
+      enabled: true
+    aws_auth:
+      region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs_awsprw/otconfig.tpl:116]:
```suggestion
      region: '${region}'
```
`awsemf` exporter uses this format. Does it need to be used here?

File: terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs_awsprw/otconfig.tpl [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        action: delete
+
+exporters:
+  awsemf:
+    namespace: '${otel_service_namespace}/${otel_service_name}'
+    region: '${region}'
+    resource_to_telemetry_conversion:
+      enabled: true
+  awsprometheusremotewrite:
+    endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
```
Comment (bryan-aguilar) [inline terraform/testcases/otlp_grpc_exporter_cw_amp_xray_ecs_awsprw/otconfig.tpl:112]:
```suggestion
    endpoint: '${cortex_instance_endpoint}/api/v1/remote_write'
```

File: terraform/testcases/otlp_metric_amp/otconfig.tpl [hunk 4]
```diff
@@ -1,6 +1,9 @@
 extensions:
   pprof:
     endpoint: 0.0.0.0:1777
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/otlp_metric_amp/otconfig.tpl:5]:
```suggestion
    region: '${region}'
```

File: terraform/testcases/otlp_metric_amp_awsprw/otconfig.tpl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+exporters:
+  logging:
+    loglevel: debug
+  awsprometheusremotewrite:
+    endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+    aws_auth:
+      region: ${region}
+      service: "aps"
+    timeout: 10s
```
Comment (bryan-aguilar) [inline terraform/testcases/otlp_metric_amp_awsprw/otconfig.tpl:21]: Do we need `''` ticks?
```suggestion
  awsprometheusremotewrite:
    endpoint: '${cortex_instance_endpoint}/api/v1/remote_write'
    aws_auth:
      region: '${region}'
      service: "aps"
    timeout: 10s
```

File: terraform/testcases/prometheus_mock/otconfig.tpl [hunk 6]
```diff
@@ -1,3 +1,6 @@
+extensions:
+  sigv4auth:
+    region: "us-west-2"
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_mock/otconfig.tpl:3]: Why is region fixed here but passed in on other configs?

File: terraform/testcases/prometheus_mock_awsprw/otconfig.tpl [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        scrape_interval: 15s
+      scrape_configs:
+      - job_name: "test-pipeline-job"
+        static_configs:
+        - targets: [ ${sample_app_listen_address_host}:${sample_app_listen_address_port} ]
+exporters:
+  awsprometheusremotewrite:
+    endpoint: "https://${mock_endpoint}"
+    aws_auth:
+      region: "us-west-2"
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_mock_awsprw/otconfig.tpl:14]: Same question on why this region is fixed.

File: terraform/testcases/prometheus_sd/otconfig.tpl [hunk 8]
```diff
@@ -1,3 +1,7 @@
+extensions:
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_sd/otconfig.tpl:3]:
```suggestion
    region: '${region}'
```

File: terraform/testcases/prometheus_sd_adot_operator/otconfig.tpl [hunk 9]
```diff
@@ -1,3 +1,7 @@
+extensions:
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_sd_adot_operator/otconfig.tpl:3]:
```suggestion
    region: '${region}'
```

File: terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl [hunk 10]
```diff
@@ -0,0 +1,31 @@
+receivers:
+      prometheus:
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl:2]: Is this tab spacing correct?

File: terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+              action: keep
+              regex: "aoc-ns-${testing_id}"
+            - source_labels: [ __meta_kubernetes_pod_label_app ]
+              action: keep
+              regex: "sample-app"
+    exporters:
+      awsprometheusremotewrite:
+        endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+        aws_auth:
+          region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl:19]:
```suggestion
      awsprometheusremotewrite:
        endpoint: '${cortex_instance_endpoint}/api/v1/remote_write'
        aws_auth:
          region: '${region}'
```

File: terraform/testcases/prometheus_sd_awsprw/otconfig.tpl [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+          regex: "aoc-ns-${testing_id}"
+        - source_labels: [ __meta_kubernetes_pod_label_app ]
+          action: keep
+          regex: "sample-app"
+exporters:
+  awsprometheusremotewrite:
+    endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+    aws_auth:
+      region: ${region}
+      service: "aps"
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_sd_awsprw/otconfig.tpl:20]: Should this use prometheus remote write and sigv4 instead? Or is this test specifically for `awsprw`?

File: terraform/testcases/prometheus_static/otconfig.tpl [hunk 13]
```diff
@@ -1,3 +1,7 @@
+extensions:
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_static/otconfig.tpl:3]:
```suggestion
    region: '${region}'
```

File: terraform/testcases/prometheus_static_adot_operator/otconfig.tpl [hunk 14]
```diff
@@ -1,3 +1,7 @@
+extensions:
+  sigv4auth:
+    region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_static_adot_operator/otconfig.tpl:3]:
```suggestion
    region: '${region}'
```

File: terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl [hunk 15]
```diff
@@ -0,0 +1,24 @@
+receivers:
+      prometheus:
+        config:
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl:3]: Is this tab spacing correct?

File: terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+            scrape_interval: 15s
+          scrape_configs:
+          - job_name: "test-prometheus-sample-app"
+            static_configs:
+            - targets: [ ${sample_app_listen_address_host}:${sample_app_listen_address_port} ]
+    exporters:
+      awsprometheusremotewrite:
+        endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+        aws_auth:
+          region: ${region}
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl:14]:
```suggestion
      awsprometheusremotewrite:
        endpoint: '${cortex_instance_endpoint}/api/v1/remote_write'
        aws_auth:
          region: '${region}'
```

File: terraform/testcases/prometheus_static_awsprw/otconfig.tpl [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+      scrape_configs:
+      - job_name: "test-prometheus-sample-app"
+        static_configs:
+        - targets: [ ${sample_app_listen_address_host}:${sample_app_listen_address_port} ]
+exporters:
+  awsprometheusremotewrite:
+    endpoint: ${cortex_instance_endpoint}/api/v1/remote_write
+    aws_auth:
+      region: ${region}
+      service: "aps"
```
Comment (bryan-aguilar) [inline terraform/testcases/prometheus_static_awsprw/otconfig.tpl:15]: Same question as before, should this be using `awsprw`?

File: terraform/testcases/otlp_metric_amp_awsprw/otconfig.tpl (hunk 5 above)
Comment (erichsueh3) [reply terraform/testcases/otlp_metric_amp_awsprw/otconfig.tpl:21]: We don't use ticks, here or above/below; any of the tests with the `_awsprw` suffix are the same as tests that already exist in the repository (and thus have been passing already)

File: terraform/testcases/prometheus_mock_awsprw/otconfig.tpl (hunk 7 above)
Comment (erichsueh3) [reply terraform/testcases/prometheus_mock_awsprw/otconfig.tpl:14]: Good question, I didn't write the original tests so I'm not sure why it's fixed, and same for above.

File: terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl (hunk 10 above)
Comment (erichsueh3) [reply terraform/testcases/prometheus_sd_adot_operator_awsprw/otconfig.tpl:2]: It does look off, but again I did not write these and are exactly what they are in the current repository

File: terraform/testcases/prometheus_sd_awsprw/otconfig.tpl (hunk 12 above)
Comment (erichsueh3) [reply terraform/testcases/prometheus_sd_awsprw/otconfig.tpl:20]: any of the tests with the `_awsprw` suffix are the same as tests that already exist in the repository (for example, the `prometheus_sd_awsprw` testcase is the same as the `prometheus_sd` testcase that lives in the repository already. I've modified the `prometheus_sd` testcase to use prw+sigv4, and moved usage of the aws prw to the `prometheus_sd_awsprw` testcase

File: terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl (hunk 15 above)
Comment (erichsueh3) [reply terraform/testcases/prometheus_static_adot_operator_awsprw/otconfig.tpl:3]: See this [comment](https://github.com/aws-observability/aws-otel-test-framework/pull/606#discussion_r840787649) above

File: terraform/testcases/prometheus_static_awsprw/otconfig.tpl (hunk 17 above)
Comment (erichsueh3) [reply terraform/testcases/prometheus_static_awsprw/otconfig.tpl:15]: See [here](https://github.com/aws-observability/aws-otel-test-framework/pull/606#discussion_r840788037)

**Comment by vasireddy99** [comment]
hey @erichsueh3, I would prefer to just add the tests specific to `prometheusremotewriteexporter+sigv4 extension` and leave the current tests uninterrupted. which would be a vice-versa in this case.

**Review by vasireddy99 — COMMENTED** [review]
hey @erichsueh3, I would prefer to just add the tests specific to prometheusremotewriteexporter+sigv4 extension and leave the current tests uninterrupted. which would be a vice-versa in this case :-), if that sounds good.

**Comment by erichsueh3** [comment]
> hey @erichsueh3, I would prefer to just add the tests specific to `prometheusremotewriteexporter+sigv4 extension` and leave the current tests uninterrupted. which would be a vice-versa in this case :-), if that sounds good.

The reason why I did it the way it is is because we recommend use of the PRW Exporter + Sigv4 Extension over the AWS PRW Exporter and thought that this was better reflected the way I structured the tests; if anyone else feels strongly about changing it to be the other way around, I can change it

**Comment by bryan-aguilar** [comment]
Do we lose any test cases with the current implementation? As long as we are confident that all the existing `aws_pwr` tests still exist, even if they are renamed, then I think it is just a semantics issue. If the current PR has removed test cases from `aws_pwr` then those need to be added back. 

It does make sense that if we are recomending the use of `prw + sigv4` then our prometheus test cases should use that. I just want to make sure that no coverage was lost from `aws_prw`.

**Comment by vasireddy99** [comment]
> > hey @erichsueh3, I would prefer to just add the tests specific to `prometheusremotewriteexporter+sigv4 extension` and leave the current tests uninterrupted. which would be a vice-versa in this case :-), if that sounds good.
> 
> The reason why I did it the way it is is because we recommend use of the PRW Exporter + Sigv4 Extension over the AWS PRW Exporter and thought that this was better reflected the way I structured the tests; if anyone else feels strongly about changing it to be the other way around, I can change it

I totally agree, but the recommendations has nothing to do with the existing tests. so by added them as seperate tests, it would be better to track `prw+sigv4` tests in our CI too, also considering the scenario that where we were not able to reproduce all the `test-cases` locally. I strongly recommend for these above reasons.

**Comment by bryan-aguilar** [comment]
> > > hey @erichsueh3, I would prefer to just add the tests specific to `prometheusremotewriteexporter+sigv4 extension` and leave the current tests uninterrupted. which would be a vice-versa in this case :-), if that sounds good.
> > 
> > 
> > The reason why I did it the way it is is because we recommend use of the PRW Exporter + Sigv4 Extension over the AWS PRW Exporter and thought that this was better reflected the way I structured the tests; if anyone else feels strongly about changing it to be the other way around, I can change it
> 
> I totally agree, but the recommendations has nothing to do with the existing tests. so it would be better to track `prw+sigv4` tests in our CI too, also considering the scenario that where we were not able to reproduce the tests locally. I strongly recommend for these above reasons.

In that case shouldn't this test case `terraform/testcases/prometheus_sd/otconfig.tpl` be renamed to `prometheus_sd_awsprw` since it uses `awsprw` instead of `prw+sigv4`. Which is the change that was already made. I think leaving prometheus test names without differentiation between exporters would be a bad choice. 

Would a compromise be to change all prometheus tests to indicate which exporter they are using? So we can have a differentiation between `sigv4` and `awsprw` implementations? In that case the above mentioned test would need some indicator that it was using `sigv4` in the name. 

The downside to this is that the additional context of `sigv4` vs `awsprw` could probably be removed from test names when `awsprw` is removed from the ADOT Collector. Would it just make sense to leave test cases that use `prw+sigv4` without any additional tag and instead tag `awsprw` tests?

**Comment by bryan-aguilar** [comment]
@erichsueh3  could you possibly provide some more context on how you renamed/moved tests in the PR description?

**Comment by erichsueh3** [comment]
> @erichsueh3 could you possibly provide some more context on how you renamed/moved tests in the PR description?

Yeah, for the `_awsprw` tests I created a new folder with the suffix for that test, went to the corresponding test in the repo, and just did a "Copy raw contents" for the `.tpl` and `tf.vars` files to make sure that those were the same. When I did this, the changes to the original test cases were already made since the recommendation to have the tests split weren't until after the PR was made

**Comment by bryan-aguilar** [comment]
That makes a lot of sense. I think this is an okay way forward. I understand wanting to be cautious about adjusting existing tests. This should be a nonissue as long the test case matrix is properly updated in the `aws-otel-collector` repository. This way makes it easier to simply delete the `_awspr` tests when `awspwr` is removed from `aws-otel-collector`.

**Review by Aneurysm9 — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

**Review by vasireddy99 — APPROVED** [review]

**Pull request merged**